core/cmd/remote_client.go

```diff
@@ -542,3 +547,28 @@ func parseResponse(resp *http.Response) ([]byte, error) {
 	}
 	return b, err
 }
+
+func (cli *Client) checkRemoteBuildCompatibility() error {
+	resp, err := cli.HTTP.Get("/v2/build_info")
+	if err != nil {
+		return cli.errorOut(err)
+	}
+	b, err := parseResponse(resp)
+	if err != nil {
+		return errors.Wrap(err, "parseResponse error")
+	}
+
+	var remote_build_info map[string]string
+	if err := json.Unmarshal(b, &remote_build_info); err != nil {
+		return errors.Wrap(err, "parseResponse error")
+	}
+	remote_version, remote_sha := remote_build_info["version"], remote_build_info["commitSHA"]
+	cli_version, cli_sha := static.Version, static.Sha
+
+	if remote_version != cli_version || remote_sha != cli_sha {
```

Comment:
We may need to consider `""` and `"unset"` as special cases here to bypass the check.

Reply:
I'm also wondering if we'll need a general way of opting out for special cases (or relaxing to semver compatibility :thinking: ...when `CHAINLINK_DEV=false`?)

Reply:
Is there a way we can log this result somewhere? We can see the usage pattern that is occurring today and then take a decision?

Reply:
In the ticket, there was a comment that this seemed like a good idea to enforce in general: https://app.shortcut.com/chainlinklabs/story/28555/cli-node-version-detection#activity-29502